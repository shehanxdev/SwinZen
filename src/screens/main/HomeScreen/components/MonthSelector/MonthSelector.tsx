import React from 'react';
import { View } from 'react-native';

import { MoveLeftArrowIcon, MoveRightArrowIcon, Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextVariant } from '@sz/constants';

import { ArrowContainer } from './components/ArrowContainer';

interface MonthSelectorProps {
  disabled?: boolean; //This prop is use to handle the edge case for free tier users
}

export function MonthSelector({ disabled = false }: MonthSelectorProps) {
  return (
    <View style={tw`flex flex-row h-18 justify-between items-center px-4`}>
      <ArrowContainer
        onArrowIconPress={() => {
          // TODO:: Implement
        }}>
        <MoveLeftArrowIcon />
      </ArrowContainer>
      <Text variant={TextVariant.SubTitle2SemiBold} color={disabled ? Color.Neutral.Sz600 : ('#1A5C23' as Color)}>
        JUNE 2022
      </Text>
      <ArrowContainer
        onArrowIconPress={() => {
          // TODO::Implement
        }}>
        <MoveRightArrowIcon />
      </ArrowContainer>
    </View>
  );
}
