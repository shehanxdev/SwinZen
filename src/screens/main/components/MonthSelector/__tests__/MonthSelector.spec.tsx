import { render } from '@testing-library/react-native';
import * as React from 'react';

import { MonthSelector } from '../MonthSelector';

describe('MonthSelector component', () => {
  const testID = 'MonthSelectorComponentTestID';

  const getRenderedComponent = () => render(<MonthSelector />);

  beforeEach(() => {
    jest.useFakeTimers().setSystemTime(new Date('2023-05-30'));
  });

  it(`should render MonthSelector component correctly`, () => {
    const renderer = getRenderedComponent();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the MonthSelector component via testID', () => {
    const { getByTestId } = getRenderedComponent();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
