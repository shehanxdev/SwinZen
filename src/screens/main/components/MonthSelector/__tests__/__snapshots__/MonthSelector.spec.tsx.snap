// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthSelector component should render MonthSelector component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 72,
      "justifyContent": "space-between",
      "paddingLeft": 16,
      "paddingRight": 16,
    }
  }
  testID="MonthSelectorComponentTestID"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1A5C23",
        "borderRadius": 9999,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
    testID="ArrowContainerComponentTestID"
  >
    <SvgMock
      fill="#FFFFFF"
      height={15}
      width={15}
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#1A5C23",
          "lineHeight": 28,
        },
        {
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 24,
          "fontWeight": "600",
          "textAlign": "center",
        },
      ]
    }
  >
    May 2023
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#6C757D",
        "borderRadius": 9999,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
    testID="ArrowContainerComponentTestID"
  >
    <SvgMock
      fill="#FFFFFF"
      height={15}
      width={15}
    />
  </View>
</View>
`;
